cpp/src/arrow/io/file.cc

```diff
@@ -207,6 +224,8 @@ class OSFile {
   FileDescriptor fd_;
   FileMode::type mode_;
   int64_t size_{-1};
+  bool reuse_;
+  int64_t offset_{0};
```

Comment:
Does this need to be updated on a seek?

Reply:
I believe the offset_ is updated on a seek. Do you mean if the reuse_ should be updated too?